hyper-v: KVP test suite port from older lisa

mbivolan opened the pull request.

iamshital commented on Testscripts/Linux/KVP-AddKeyValues.sh:
```diff
@@ -0,0 +1,142 @@
+
+
+
+dos2unix utils.sh
```
Remove `dos2unix utils.sh`. File will be converted to unix before upload.

iamshital commented on Testscripts/Linux/KVP-BasicChecks.sh:
```diff
@@ -0,0 +1,249 @@
+    fi
+}
+
+dos2unix utils.sh
```
Remove dos2unix utils.sh. File will be converted to unix before upload.

iamshital commented on Libraries/Test-Framework.psm1:
```diff
@@ -454,10 +446,15 @@ function Run-Test {
-            $AllVMData = Check-IP -VMData $AllVMData
-            Set-Variable -Name AllVMData -Value $AllVMData -Scope Global
-            LogMsg "Public IP found for all VMs in deployment after checkpoint restore"
+            if ($CurrentTestData.applyCheckpoint -eq "False") {
```
Please put `applyCheckpoint` into `AdditionalHWConfig.HyperVApplyCheckpoint` to make it consistent.
Example: https://github.com/LIS/LISAv2/blob/master/XML/TestCases/PerformanceTests.xml#L6

juhlee-microsoft commented on XML/TestCases/KVP.xml:
```diff
@@ -0,0 +1,173 @@
+        <Platform>HyperV</Platform>
+        <Category>KVP</Category>
+        <Area>KVP</Area>
+        <Tags>sqm</Tags>
```
is it worth to have separated tag? All others are kvp.

juhlee-microsoft replied:
acked

lizzha commented on Testscripts/Windows/TCUtils.ps1:
```diff
@@ -0,0 +1,2441 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
Can you follow the PowerShell naming convention that we agreed?
1.	Method name, use the fashion of verb-noun based names, UpperCamelCase, such as Test-Object. We can use Get-Verb to see what verbs are available.
2.	Global Variables: UpperCamelCase, such as UserName
3.	Local variable: lowerCamelCase, such as localFileName
4.	Const: all upper cases, such as: SCRIPT_VERSION
5.	Parameter name: UpperCamelCase

mbivolan commented:
All the review comments were addressed and appended in the original commmits:
- @iamshital:
    - removed dos2unix from sh scripts
    - changed applyCheckpoint param into HyperVApplyCheckpoint
- @lizzha:
    - replaced TCUtils.ps1 with testcase-utils.psm1 (contains only the methods required by the tests in this PR)
    - refactored all the scripts following the conventions you specified
- @juhlee-microsoft:
    - changed sqm tag name to kvp

The tests in this PR were run against the following images:
- Hyper-V:
    - CentOS 7.5: Everything PASS
    - Debian 8.10: Only KVP-TestRescing FAIL
    - Ubuntu 18.04: Only KVP-TestRescing FAIL
- Azure:
    - RHEL 7.5: KVP-BasicChecks FAIL
    - Ubuntu 16: KVP-BasicChecks PASS

iamshital commented:
@mbivolan Can you please send the test logs in email?

ader1990 commented on Libraries/CommonFunctions.psm1:
```diff
@@ -2573,4 +2573,228 @@ function CreateTestResultObject()
+    $check2 = $true
+    
+    .\Tools\plink.exe -C -pw $Password -P $SSHPort $Username@$Ipv4 "ls -l /sbin/init | grep systemd"
+    if ($? -ne "True") {
```
use lastexitcode var

ader1990 replied:
this applies to all the future occurrences

ader1990 commented on Libraries/CommonFunctions.psm1:
```diff
@@ -2573,4 +2573,228 @@ function CreateTestResultObject()
+function Get-VMFeatureSupportStatus {
+    <#
+    .Synopsis
+        Check if VM supports one feature or not.
```
a feature

ader1990 commented on Libraries/CommonFunctions.psm1:
```diff
@@ -2573,4 +2573,228 @@ function CreateTestResultObject()
+        Check if VM supports one feature or not.
+    .Description
+        Check if VM supports one feature or not based on comparison of curent kernel version with feature
+        supported kernel version. If the current version is lower than feature supported version, return false, otherwise return true.
```
line too long

ader1990 commented on Libraries/CommonFunctions.psm1:
```diff
@@ -2573,4 +2573,228 @@ function CreateTestResultObject()
+        [String] $SupportKernel
+    )
+    
+    echo y | .\Tools\plink.exe -C -pw $Password -P $SSHPort $Username@$Ipv4 'exit 0'
```
@iamshital @chvalean is there a reason not to use ssh.exe from portable git? or a manual build? 
we would not be required to always execute this line to accept the key

ader1990 commented on Testscripts/Linux/KVP-AddKeyValues.sh:
```diff
@@ -0,0 +1,130 @@
+#
+# Delete any summary.log files from a previous run
+#
+rm -f ~/summary.log
```
reuse utils.sh and not use summary.log directly
